src/compiler/checker.ts

```diff
@@ -27281,16 +27281,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return false;
     }
 
-    function isDiscriminantProperty(type: Type | undefined, name: __String) {
+    function isDiscriminantProperty(type: Type | undefined, name: __String, considerNonUniformPrimitivePropDiscriminant: boolean = true) {

```

Comment:
`considerNonUniformPrimitivePropDiscriminant` could use a comment above this function that would explain its purpose and how it's used